transport/server.cc

```diff
@@ -238,14 +238,21 @@ cql_server::do_accepts(int which, bool keepalive, socket_address server_addr) {
             auto conn = make_shared<connection>(*this, server_addr, std::move(fd), std::move(addr));
             ++_connects;
             ++_connections;
-            // Move connection to the background, monitor for lifetime and errors.
-            (void)conn->process().then_wrapped([this, conn] (future<> f) {
-                --_connections;
+            // Move conn to the background, announce, then block while monitoring for lifetime/errors.
+            (void)futurize_apply([this, conn] {
+                return advertise_new_connection(conn);
+            }).then_wrapped([this, conn] (future<> f) {
                 try {
                     f.get();
-                } catch (...) {
-                    clogger.debug("connection error: {}", std::current_exception());
+                } catch(...) {
+                    clogger.debug("exception while advertising new connection: {}", std::current_exception());
                 }
+                return conn->process().finally([this, conn] {
+                    --_connections;
+                    return unadvertise_connection(conn);
+                }).handle_exception([] (auto ep) {
+                    clogger.debug("exception found after connection close: {}", ep);
```

Comment:
Again, info level or we'll never see it.

Please avoid gratuitus generic lambdas, they're hard for IDEs to understand.

Reply:
Both done.